TCAT/Models/WalkPath.swift

```diff
@@ -28,10 +28,38 @@ class WalkPath: Path {
         self.path = untraveledPath
         self.strokeColor = color
         self.strokeWidth = polylineWidth
+        
+        guard let path = self.path else { return }
+        let intervalDistanceIncrement: CGFloat = 20
+        var previousCircle: (coordinate: CLLocationCoordinate2D, radius: Double)?
+        // Maps circle coordinates in incremental distance
+        for coordinateIndex in 0 ..< path.count() - 1 {
+            let startCoordinate = path.coordinate(at: coordinateIndex)
+            let endCoordinate = path.coordinate(at: coordinateIndex + 1)
+            let startLocation = CLLocation(latitude: startCoordinate.latitude, longitude: startCoordinate.longitude)
+            let endLocation = CLLocation(latitude: endCoordinate.latitude, longitude: endCoordinate.longitude)
+            let pathDistance = endLocation.distance(from: startLocation)
+            let intervalLatIncrement = (endLocation.coordinate.latitude - startLocation.coordinate.latitude) / pathDistance
+            let intervalLngIncrement = (endLocation.coordinate.longitude - startLocation.coordinate.longitude) / pathDistance
+
+            for intervalDistance in 0 ..< Int(pathDistance) {
```

Comment:
Nice! This is an example where a for loop would be more useful since we are using the range operator ..<